qutebrowser/mainwindow/mainwindow.py

```
@@ -54,44 +54,59 @@ def get_window(via_ipc, force_window=False, force_tab=False,
-        window_to_raise = window
-    else:
-        try:
-            win_mode = config.get('general', 'new-instance-open-target.window')
-            if win_mode == 'last-focused':
-                window = objreg.last_focused_window()
-            elif win_mode == 'last-opened':
-                window = objreg.last_window()
-            elif win_mode == 'last-visible':
-                window = objreg.last_visible_window()
-        except objreg.NoWindow:
-            # There is no window left, so we open a new one
-            window = MainWindow()
-            window.show()
-            win_id = window.win_id
-            window_to_raise = window
-        win_id = window.win_id
-        if open_target not in ['tab-silent', 'tab-bg-silent']:
-            window_to_raise = window
-    if window_to_raise is not None:
-        window_to_raise.setWindowState(
-            window.windowState() & ~Qt.WindowMinimized | Qt.WindowActive)
-        window_to_raise.raise_()
-        window_to_raise.activateWindow()
-        QApplication.instance().alert(window_to_raise)
-    return win_id
+        raise_window = True
+
+    if raise_window:
+        window.setWindowState(window.windowState() & ~Qt.WindowMinimized)
+        window.setWindowState(window.windowState() | Qt.WindowActive)
+        window.raise_()
+        window.activateWindow()
+        QApplication.instance().alert(window)
+
+    return window.win_id
+
+
+def get_target_window():
+    """Get the target window for new tabs, or None if none exist."""
+    try:
+        win_mode = config.get('general', 'new-instance-open-target.window')
+        if win_mode == 'last-focused':
+            return objreg.last_focused_window()
+        elif win_mode == 'first-opened':
+            return objreg.window_by_index(0)
+        elif win_mode == 'last-opened':
+            return objreg.window_by_index(-1)
+        elif win_mode == 'last-visible':
+            return objreg.last_visible_window()
```

Please add an explicit `else:` here which just does `raise ValueError("Invalid win_mode {}".format(win_mode))` or so. That way, it fails when we ever add a new mode and forget to handle it here instead of implicitly returning `None`.